spec/factories/course_materials.rb

```diff
@@ -4,6 +4,11 @@
     folder
     sequence(:name) { |n| "Material #{n}" }
     sequence(:description) { |n| "Material Description #{n}" }
-    attachment
+
+    after(:build) do |material|
```

Comment:
This is because `:course_material` and `:attachment_reference` depends on each other in the factory after `null: false` is added.  

It works before only because the old attachment has a `null: false` on attachable so that the its factory does not depends on `:course_material`